### DISCUSSION TITLE
verify bucket policy and IAM policy

### DISCUSSION THREAD

**Discussion by hoyho** [minio/minio#12805] [Q&A] [answered]
As I check at the policy implementation, I find the verify function here `isPutActionAllowed` in PutObject API:
https://github.com/minio/minio/blob/471b4fd0c93e0ec7e80018273e73f775dda67561/cmd/auth-handler.go#L655
It check the bucket policy(via globalPolicySys) only when AccessKey is empty. Although it also do a check on IAM policy later.
My question is why it only check bucket policy when AccessKey is empty. 
Is there any design or detail I can find？

**Comment by harshavardhana** [comment, marked as answer]
Bucket policies are only meant for anonymous access.
